homeassistant/components/simplisafe/binary_sensor.py

```diff
@@ -0,0 +1,279 @@
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+    # Add low battery status entity for every sensor
+    async_add_entities(
+        [
+            SimpliSafeSensorBattery(simplisafe, system, sensor)
+            for system in simplisafe.systems.values()
+            for sensor in system.sensors.values()
+            if sensor.type in SUPPORTED_SENSOR_TYPES
+        ]
+    )
+
+    # Add motion sensor for cameras and doorbells
+    cameras = []
+    doorbells = []
+    for system in simplisafe.systems.values():
+        if (
+            "cameras" in system._location_info["system"]
+            and len(system._location_info["system"]["cameras"]) > 0
+        ):
+            for cam in system._location_info["system"]["cameras"]:
+                cameras.append(cam)
+                if cam["model"] == "SS002":
+                    doorbells.append(cam)
+
+    async_add_entities(
+        [
+            SimpliSafeCameraBinarySensor(
+                simplisafe, system, camera, DEVICE_CLASS_MOTION
+            )
+            for camera in cameras
+        ]
+    )
+
+    # Add occupancy sensor for doorbells
+    async_add_entities(
+        [
+            SimpliSafeCameraBinarySensor(
+                simplisafe, system, camera, DEVICE_CLASS_OCCUPANCY
+            )
+            for camera in doorbells
+        ]
+    )
+
+
+class SimpliSafeBinarySensor(SimpliSafeEntity, BinarySensorEntity):
+    """Define a SimpliSafe binary sensor entity."""
+
+    def __init__(self, simplisafe, system, sensor):
+        """Initialize."""
+        super().__init__(simplisafe, system, sensor.name, serial=sensor.serial)
+        self._system = system
+        self._sensor = sensor
+        self._is_on = False
+
+    @property
+    def device_class(self):
+        """Return type of sensor."""
+        return HA_SENSOR_TYPES[self._sensor.type]
+
+    @property
+    def unique_id(self):
+        """Return unique ID of sensor."""
+        return f"{self._sensor.serial}-sensor"
+
+    @property
+    def device_info(self):
+        """Return device registry information for this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._sensor.serial)},
+            "manufacturer": "SimpliSafe",
+            "model": SENSOR_MODELS[self._sensor.type],
+            "name": self._sensor.name,
+            "via_device": (DOMAIN, self._system.serial),
+        }
+
+    @property
+    def is_on(self):
+        """Return true if the sensor is on."""
+        return self._is_on
+
+    @callback
+    def async_update_from_rest_api(self):
+        """Update the entity with the provided REST API data."""
+        self._is_on = self._sensor.triggered
+
+    @callback
+    def async_update_from_websocket_event(self, event):
+        """Update the entity with the provided websocket event data."""
+        if (
+            event.event_type == EVENT_MOTION_DETECTED
```

Comment:
I understand that other integrations surface events like this as entities, but we already have a way to detect (and run automation on) these events: https://www.home-assistant.io/integrations/simplisafe/#events – what does your implementation provide beyond the event architecture?

Reply:
This is to surface this as a visible binary sensor in the UI, like other motion sensors. I've taken the implementation from similar sensors already implemented in HA.

Reply:
But why is that UI needed (especially since we're manually flipping the sensor 10 seconds later)? The events provide the same mechanism, but in a more flexible way (which can create notifications, etc.).